src/bot.ts

```diff
@@ -114,6 +136,55 @@ const startupChores: Chore[] = [
 		},
 		complete: "All slash commands registered",
 	},
+	{
+		summary: "Update guild slash command permissions",
+		fn: async () => {
+			const guild = await client.guilds.cache.get(GUILD_ID as string);
+			const commands = guild?.commands;
+
+			const fetched = await commands?.fetch();
+
+			if (fetched) {
+				await commands?.permissions.set({
+					fullPermissions: fetched.map((c) => {
+						let commandSpecificPermission:
+							| Discord.ApplicationCommandPermissionData
+							| undefined;
+						const perm =
+							commandPermissions[c.name] ??
+							DEFAULT_COMMAND_PERMISSION;
+						switch (perm) {
+							case CommandPermission.Admin:
+								commandSpecificPermission = {
+									id: process.env.ADMIN_ID as string,
+									type: "ROLE",
+									permission: true,
+								};
+								break;
+							case CommandPermission.ServerOwner: {
+								const owner = guild?.ownerId;
+								if (owner) {
+									commandSpecificPermission = {
+										id: owner,
+										type: "USER",
+										permission: true,
+									};
+								}
+								break;
+							}
+						}
+						return {
+							id: c.id,
+							permissions: commandSpecificPermission
+								? [commandSpecificPermission]
+								: [],
```

Comment:
I've removed the "backdoor" from here.
How are public commands handled here? Should it return an empty list or `undefined`?

Reply:
`commandSpecificPermission` isn't set (therefore `undefined`), so the ternary returns an empty array and the command's permissions are set to that empty array (no permission overrides).

Note that when being created, public commands already have their [default permission](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-structure) set to true.